### PR TITLE
Fixes #243: Clicking on the uploaded image opens in custom tabs

### LINKED ISSUE #243: Clicking on an uploaded image causes it to open in a browser
We should probably instead open it in a lightbox in-app, similar to how we handle this on web.

### PR DESCRIPTION
**Pull request by saketkumar95**
Fixes : [#243](https://github.com/zulip/zulip-android/issues/243) - Now the uploaded image opens in chrome custom tabs.

**Demo video here:** 

https://drive.google.com/file/d/0Bx-iZdT4RNS9UHZSQ1hqVnRhdTQ/view?usp=sharing

@timabbott @kunall17 Tell me if you need any changes.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@saketkumar95, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by timabbott** [comment]
Cool!  Video looks nice; @kunall17 @Sam1301 can one of you review?

File: app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java [hunk 1]
```diff
@@ -381,10 +384,20 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
...
-                                    i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                                    i.setData(Uri.parse(url));
-                                    zulipApp.startActivity(i);
+                                    if (Build.VERSION.SDK_INT < 15)
+                                    {
+                                        Intent i = new Intent(Intent.ACTION_VIEW);
+                                        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                                        i.setData(Uri.parse(url));
+                                        zulipApp.startActivity(i);
+                                    }
```
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: Here should be an else case!

Comment (saketkumar95) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: At first a i added that but it was working fine in both the versions so i just removed that. 
Well added it again. Review now. :)

**Comment by kunall17** [comment]
I tried this on my local setup, it redirected me to image but it asked me to login obviously because i was not logged from my chrome browser! 
And in the zulip.tabbott.net the server allows you to browse the uploads without login! 
@timabbott Am I right with these assumptions? 

Also Squash the commit eaa4f70! 
And also consider installing a dev env it is fairly easy! :)

**Comment by saketkumar95** [comment]
@kunall17 Yeah dev env is installed. I just want to check on both so i did. :)

File: app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java [hunk 2]
```diff
@@ -381,10 +384,22 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
...
+                                    else
+                                    {
+                                        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
+                                        builder.setStartAnimations(zulipApp.getZulipActivity(), R.anim.slide_in_right, R.anim.slide_out_left);
```
Comment (vishwesh3) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: @saketkumar95 You can use `android.R.anim.slide_in_left` , `android.R.anim.slide_out_right` . By using android inbuilt transition files we don't have to create anim xml and will eventually reduce APK size little bit.

**Comment by saketkumar95** [comment]
@kunall17 Done!

File: app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java (hunk 2 above)
Comment (kunall17) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: @saketkumar95 this one is left! 
And resolve the merge conflicts

Comment (saketkumar95) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: @kunall17 I tried this but it was not working in chrome custom tabs when i click the the exit button.

Comment (saketkumar95) [inline app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java]: @vishwesh3 android inbuit transition don't have transition like `android.R.anim.slide_in_right` and `android.R.anim.slide_out_left` . So eventually i need this anim xml file to do so. 
If i'm missing out something then do correct me. :)

**Comment by saketkumar95** [comment]
@kunall17 Everthing is done. Do i need to send the PR of Download and copy link feature to this PR only? Or should i create another PR for that??

**Pull request closed without merging**